### DISCUSSION TITLE
Signed off commits show email to the public

### DISCUSSION THREAD

**Discussion by Tonipetrov** [community/community#23987] [New to GitHub]
<p>Hello,</p>
<p>We have a GitHub organization and we would like to add a requirement when external contributors commit code, that they will need to add -s in order to signoff (signoff will mean that they agree to our contributor agreement).</p>
<p>But while testing we were surprised to see that the userid and email address become public in the commit (even anonymous users can see them. That seems to be a problem - some people will not want their email address to be public and available to crawlers, etc. And having company userid-s public is even worse (I think it will be against many companies policies)…</p>
<p>Is there a way to hide these for users who are not admins of the organization and the person who commited (I expected that would be the way it works) for public repositories? Otherwise this seems too violating…</p>

**Comment by byrneh** [comment, marked as answer]
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener nofollow ugc">Setting your commit email address - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<p>
GitHub suggestion in their docs link above is</p>
<p>If you’d like to keep your personal email address private, you can use a GitHub-provided <code>no-reply</code> email address as your commit email address. To use your <code>noreply</code> email address for commits you push from the command line, use that email address when you set your commit email address in Git. To use your <code>noreply</code> address for web-based Git operations, set your commit email address on GitHub and choose to <strong>Keep my email address private</strong> .</p>

**Comment by airtower-luna** [comment]
<p>Sign-off lines are part of the commit message, so they will be available to anyone with read access to the repository. That’s a general Git thing, not just a property of the GitHub website.</p>
<p>If you want to avoid that you’ll have to discuss with your legal department/lawyer</p>
<ul>
<li>whether you really need that sign-off, and if so</li>
<li>whether the noreply address that <a class="mention" href="/byrneh">@byrneh</a> mentioned would be acceptable for sign-off for people who’d rather keep their name and mail address private.</li>
</ul>
